### PR TITLE
Issue 213

### LINKED ISSUE #213: Feature based on similar names
Extract a similar-names feature from the dataset pair, to be implemented in workflow.extract_features.

It should be inspired by the baseline similar names match.

Other features implemented in `feature_extraction.py`

Useful docs:
https://recordlinkage.readthedocs.io/en/latest/ref-compare.html
https://recordlinkage.readthedocs.io/en/latest/ref-compare.html#user-defined-algorithms

### PR DESCRIPTION
**Pull request by MaxFrax**
Closes #213

### CODE REVIEW

File: soweego/linker/feature_extraction.py [hunk 1]
```diff
@@ -290,3 +290,40 @@ def _pair_has_any_null(pair):
         return True
 
     return False
+
+
+class SimilarTokens(BaseCompareFeature):
+    name = 'SimilarTokens'
+    description = 'Compare pairs of lists with URL values'
```
Comment (marfox) [inline soweego/linker/feature_extraction.py]:
```suggestion
    description = 'Compare pairs of lists with string values based on shared tokens'
```

File: soweego/linker/feature_extraction.py [hunk 2]
```diff
@@ -290,3 +290,40 @@ def _pair_has_any_null(pair):
...
+    def __init__(self, left_on, right_on, agree_value=1.0, disagree_value=0.0, missing_value=0.0, label=None):
+        super(SimilarTokens, self).__init__(left_on, right_on, label=label)
+        self.agree_value = agree_value
+        self.disagree_value = disagree_value
+        self.missing_value = missing_value
+
+    def _compute_vectorized(self, source_column, target_column):
+        concatenated = pd.Series(list(zip(source_column, target_column)))
+
+        def exact_apply(pair):
```
[the remark this answers is not in the document]
Comment (MaxFrax) [reply soweego/linker/feature_extraction.py]: There's no logic behind it. I wasn't getting too much when I copy-pasted the Url feature code.
I'm going to change it right away.

File: soweego/linker/feature_extraction.py [hunk 3]
```diff
@@ -290,3 +290,40 @@ def _pair_has_any_null(pair):
...
+
+            source_labels = [pair[0]] if isinstance(pair[0], str) else pair[0]
+            target_labels = [pair[1]] if isinstance(pair[1], str) else pair[1]
+
+            first_set = set(source_labels)
+            second_set = set()
+
+            for label in target_labels:
+                if label:
+                    second_set.update(filter(None, text_utils.tokenize(label)))
```
[the remark this answers is not in the document]
Comment (MaxFrax) [reply soweego/linker/feature_extraction.py]: You have been tricked by the arguments names. 
When this code is executed, the `target_labels` are filled with wikidata values.

Probably I should change the arguments name.

File: soweego/linker/feature_extraction.py [hunk 4]
```diff
@@ -290,3 +290,40 @@ def _pair_has_any_null(pair):
...
+
+            source_labels = [pair[0]] if isinstance(pair[0], str) else pair[0]
+            target_labels = [pair[1]] if isinstance(pair[1], str) else pair[1]
+
+            first_set = set(source_labels)
+            second_set = set()
+
+            for label in target_labels:
+                if label:
+                    second_set.update(filter(None, label.split(' ')))
```
Comment (marfox) [inline soweego/linker/feature_extraction.py]:
```suggestion
                    second_set.update(filter(None, label.split()))
```

**Pull request merged**